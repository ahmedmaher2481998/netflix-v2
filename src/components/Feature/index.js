import { Container, Break, Title, SubTitle } from './styles/Feature';
import React from 'react';

const Feature = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Feature.Title = function FeatureTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Feature.Break = function FeatureBreak({ children, ...restProps }) {
  return <Break {...restProps}>{children}</Break>;
};

Feature.SubTitle = function FeatureSubTitle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

export default Feature;
