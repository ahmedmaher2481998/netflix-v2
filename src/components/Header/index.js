import { Background, ButtonLink, Frame, Logo } from './styles/Header';

const Header = ({ bg, Children, ...restProps }) => {
  return bg ? <Background {...restProps}>{Children}</Background> : Children;
};

Header.Logo = function HeaderLogo({ children, ...restProps }) {
  return <Logo {...restProps}>{children}</Logo>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Frame {...restProps}>{children}</Frame>;
};
Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

export default Header;
