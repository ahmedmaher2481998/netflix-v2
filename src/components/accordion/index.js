import { createContext, useContext, useState } from 'react';
import { Container, Inner, Body, Frame, Header, Item, Title } from './styles/accordion';

const ShowToggleContext = createContext();

function Accordion({ children, ...resProps }) {
  return (
    <>
      <Container>
        <Inner {...resProps}>{children}</Inner>
      </Container>
    </>
  );
}

Accordion.Item = function AccordionItem({ children, ...resProps }) {
  const [showToggle, setShowToggle] = useState(false);

  return (
    <ShowToggleContext.Provider value={{ showToggle, setShowToggle }}>
      <Item {...resProps}> {children}</Item>
    </ShowToggleContext.Provider>
  );
};

Accordion.Frame = function AccordionFrame({ children, ...resProps }) {
  return <Frame {...resProps}> {children}</Frame>;
};

Accordion.Header = function AccordionHeader({ children, ...resProps }) {
  const { setShowToggle, showToggle } = useContext(ShowToggleContext);
  return (
    <Header onClick={() => setShowToggle((pre) => !pre)} {...resProps}>
      {children}
      {showToggle ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

Accordion.Body = function AccordionBody({ children, ...resProps }) {
  const { showToggle } = useContext(ShowToggleContext);
  return showToggle ? <Body {...resProps}> {children}</Body> : null;
};

Accordion.Title = function AccordionTitle({ children, ...resProps }) {
  return <Title {...resProps}> {children}</Title>;
};

export default Accordion;
