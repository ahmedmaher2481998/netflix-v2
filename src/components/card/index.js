import React from 'react';
import {
  Container,
  Title,
  SubTitle,
  Item,
  Image,
  Feature,
  FeatureTitle,
  FeatureText,
  FeatureClose,
  Group,
  Text,
  Entities,
  Meta,
  Maturity,
  Content
} from './styles/cards';
import { useContext, createContext, useState } from 'react';
const FeatureContext = createContext();
const Card = ({ children, ...rest }) => {
  const [showFeature, setShowFeature] = useState(false);
  const [itemFeature, setItemFeature] = useState({});
  return (
    <FeatureContext.Provider
      value={{
        showFeature,
        setShowFeature,
        itemFeature,
        setItemFeature
      }}></FeatureContext.Provider>
  );
};

export default Card;
