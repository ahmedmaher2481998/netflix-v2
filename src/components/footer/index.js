import React from "react";
import {
	Title,
	Row,
	Column,
	Container,
	Link,
	Text,
	Break,
} from "./styles/footer";
const Footer = () => {
	return <div>Footer</div>;
};
Footer.Container = ({ children, ...resProps }) => (
	<Container {...resProps}>{children}</Container>
);

Footer.Title = ({ children, ...resProps }) => (
	<Title {...resProps}> {children}</Title>
);
Footer.Row = ({ children, ...resProps }) => (
	<Row {...resProps}> {children}</Row>
);

Footer.Column = ({ children, ...resProps }) => (
	<Column {...resProps}> {children}</Column>
);

Footer.Link = ({ children, ...resProps }) => (
	<Link {...resProps}> {children}</Link>
);

Footer.Text = ({ children, ...resProps }) => (
	<Text {...resProps}> {children}</Text>
);

Footer.Break = ({ children, ...resProps }) => (
	<Break {...resProps}> {children}</Break>
);

export default Footer;
