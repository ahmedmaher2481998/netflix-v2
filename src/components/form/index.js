import React from 'react';
import { Link, Error, Container, Text, SmallText, Input, Submit, Title, Base } from './styles/form';
const Form = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.SmallText = function FormSmallText({ children, ...restProps }) {
  return <SmallText {...restProps}>{children}</SmallText>;
};

Form.Input = function FormInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Link = function FormLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

export default Form;
