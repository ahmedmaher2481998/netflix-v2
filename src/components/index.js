export { default as JumboTron } from './jumbotron/index';
export { default as Footer } from './footer/index';
export { default as Accordion } from './accordion/index';
