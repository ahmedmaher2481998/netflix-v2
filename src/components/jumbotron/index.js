import React from 'react';
import { Inner, Item, Pane, Container, Image, Subtitle, Title } from './styles/jumbotron';

function JumboTron({ children, direction = 'row', ...restProps }) {
  return (
    <Item {...restProps}>
      <Inner direction={direction}>{children}</Inner>
    </Item>
  );
}

JumboTron.Container = function JumboTronContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

JumboTron.Pane = function JumboTronPane({ children, ...restProps }) {
  return <Pane {...restProps}>{children}</Pane>;
};

JumboTron.Title = function JumboTronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

JumboTron.Subtitle = function JumboTronSubtitle({ children, ...restProps }) {
  return <Subtitle {...restProps}>{children}</Subtitle>;
};

JumboTron.Image = function JumboTronImage({ ...restProps }) {
  return <Image {...restProps} />;
};

export default JumboTron;
