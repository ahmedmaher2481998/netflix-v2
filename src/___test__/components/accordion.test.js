import { fireEvent, render } from '@testing-library/react';
import { Accordion } from '../../components';
import FAQData from '../../fixtures/faqs.json';

describe('<Accordion>', () => {
  it('Renders <Accordion/> with populated data', () => {
    const { container, getByText } = render(
      <>
        <Accordion>
          <Accordion.Frame>
            <Accordion.Title>Frequently Asked Questions</Accordion.Title>
            {FAQData.map((item) => {
              return (
                <Accordion.Item key={item.id}>
                  <Accordion.Header> {item.header} </Accordion.Header>
                  <Accordion.Body>{item.body}</Accordion.Body>
                </Accordion.Item>
              );
            })}
          </Accordion.Frame>
        </Accordion>
      </>
    );

    expect(getByText('Frequently Asked Questions')).toBeTruthy();
    expect(getByText(FAQData[0].header)).toBeTruthy();
    expect(getByText(FAQData[1].header)).toBeTruthy();
    expect(getByText(FAQData[2].header)).toBeTruthy();
    expect(getByText(FAQData[3].header)).toBeTruthy();
    expect(getByText(FAQData[4].header)).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('toggles the body of the <Accordion.Item/> on click', () => {
    const { container, queryByText, getByTestId } = render(
      <>
        <Accordion>
          <Accordion.Title>Frequently Asked Questions</Accordion.Title>
          <Accordion.Frame>
            {FAQData.map((item) => {
              return (
                <Accordion.Item key={item.id}>
                  <Accordion.Header> {item.header} </Accordion.Header>
                  <Accordion.Body>{item.body}</Accordion.Body>
                </Accordion.Item>
              );
            })}
          </Accordion.Frame>
        </Accordion>
      </>
    );
    const whatIsNetflixBodyText =
      "Netflix is a streaming service that offers a wide variety of award-winning TV programmes, films, anime, documentaries and more – on thousands of internet-connected devices. You can watch as much as you want, whenever you want, without a single advert – all for one low monthly price. There's always something new to discover, and new TV programmes and films are added every week!";

    expect(queryByText(whatIsNetflixBodyText)).toBeFalsy();
    fireEvent.click(queryByText('What is Netflix?'));
    expect(queryByText(whatIsNetflixBodyText)).toBeTruthy();

    fireEvent.click(queryByText('What is Netflix?'));
    expect(queryByText(whatIsNetflixBodyText)).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
