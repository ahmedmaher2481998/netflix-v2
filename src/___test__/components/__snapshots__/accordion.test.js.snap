// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion> Renders <Accordion/> with populated data 1`] = `
<div
  class="accordion__Container-sc-1owlhjn-0 eyIHCQ"
>
  <div
    class="accordion__Inner-sc-1owlhjn-2 dYMCgV"
  >
    <div
      class="accordion__Frame-sc-1owlhjn-1 kiMetJ"
    >
       
      <h1
        class="accordion__Title-sc-1owlhjn-3 bxPBqH"
      >
         
        Frequently Asked Questions
      </h1>
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          What is Netflix?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          How much does Netflix cost?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          Where can I watch?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          How do I cancel?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          What can I watch on Netflix?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion> toggles the body of the <Accordion.Item/> on click 1`] = `
<div
  class="accordion__Container-sc-1owlhjn-0 eyIHCQ"
>
  <div
    class="accordion__Inner-sc-1owlhjn-2 dYMCgV"
  >
    <h1
      class="accordion__Title-sc-1owlhjn-3 bxPBqH"
    >
       
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Frame-sc-1owlhjn-1 kiMetJ"
    >
       
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          What is Netflix?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          How much does Netflix cost?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          Where can I watch?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          How do I cancel?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          What can I watch on Netflix?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion> toggles the body of the item on click 1`] = `
<div
  class="accordion__Container-sc-1owlhjn-0 eyIHCQ"
>
  <div
    class="accordion__Inner-sc-1owlhjn-2 dYMCgV"
  >
    <div
      class="accordion__Frame-sc-1owlhjn-1 kiMetJ"
    >
       
      <h1
        class="accordion__Title-sc-1owlhjn-3 bxPBqH"
      >
         
        Frequently Asked Questions
      </h1>
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          What is Netflix?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          How much does Netflix cost?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          Where can I watch?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          How do I cancel?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
      <div
        class="accordion__Item-sc-1owlhjn-4 fPKBwt"
      >
         
        <div
          class="accordion__Header-sc-1owlhjn-5 jLiUnx"
        >
           
          What can I watch on Netflix?
           
          <img
            alt="Open"
            src="/images/icons/add.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
